docs/writing_tests.md

```diff
@@ -49,14 +49,30 @@ See documentation comments for a full description of the available API. Here is
+If an assertion fails, you may wish to either continue and allow more failures to accumulate, or immediately stop the test. That is up to you based on whether the failure is significant enough that it would make the rest of the test moot. The `ldtest.T` method `Errorf` records a failure without stopping, and `FailNow` makes it immediately terminate. When using the `testify` packages, the `assert` functions will call only `Errorf` whereas the `require` functions will call both `Errorf` and `FailNow`, so `require` always implies "stop here on failure". The `matchers` API follows the same conventions with its `Assert` and `Require` methods.
 
 ## Making test failures clear
 
 If an SDK test fails, someone will have to interpret the test output and find the problem. They should not need to dig through the test harness code to understand the failure message (although the output does include a stacktrace to make that possible). So, if for instance the assertion is that the `variationIndex` property in some event data should have value X, ideally a failed assertion would not just print "expected value X but was Y", but should include some context to clarify that this is a `variationIndex`.
 
-There are several ways to do this. When using `testify/assert` or `testify/require`, the assertion functions have optional parameters for an additional message to be printed on failure. 
\ No newline at end of file
+There are several ways to do this. When using `testify/assert` or `testify/require`, the assertion functions have optional parameters for an additional message to be printed on failure.
+
+Also, using the `matchers` API sometimes results in more helpful error messages. For instance, if you want to assert that every element in a slice has a certain property, using `matchers.Items` or `matchers.ItemsInAnyOrder` along with a matcher for the desired property is better than doing an equality test on the whole slice against the entire expected data set, because it won't care about any irrelevant properties, and because the property matcher will automatically give you an appropriate failure message. It's also better than doing a `for` loop over the slice and doing an assertion on each element, because you won't have to construct your own message to explain that the failure is for item 0, item 1, etc.
+
+Values in failure output may be formatted differently in `matchers`versus `testify`: for instance, values of types that have a JSON representation are shown as JSON in `matchers`, rather than using the result of `fmt.Sprintf("%v", value)` as `testify` does in most cases. The easiest way to see which kind of output would be most helpful for a particular test is to cause a deliberate failure in the test and see what you get.
```

Comment:
```suggestion
Values in failure output may be formatted differently in `matchers` versus `testify`: for instance, values of types that have a JSON representation are shown as JSON in `matchers`, rather than using the result of `fmt.Sprintf("%v", value)` as `testify` does in most cases. The easiest way to see which kind of output would be most helpful for a particular test is to cause a deliberate failure in the test and see what you get.
```